### PR TITLE
[TA-3386]: NewWalletFromMnemonic

### PR DESCRIPTION
**Pull request by GuillemGarciaDev**
# [TA-3386]: NewWalletFromMnemonic

This PR aims to add a new wallet method `NewWalletFromMnemonic` method.

### address-codec
- Fixed secp256k1 private key prefix

### keypairs
- Add secp256k1 crypto algorithm

### xrpl
- Add new method `NewWalletFromMnemonic`

### CODE REVIEW

File: .github/workflows/xrpl-go-ci.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ on:
   push:
     branches: [ main ]
   pull_request:
-    branches: [ main, 'v*' ]
+    branches: [ '**' ]
```
Comment (florent-uzio) [inline .github/workflows/xrpl-go-ci.yml:6]: What does `**` mean?

File: xrpl/wallet.go [hunk 2]
```diff
@@ -65,55 +70,55 @@ func NewWalletFromSecret(seed string) (Wallet, error) {
...
+
+	// Derive classic address
+	classicAddr, err := keypairs.DeriveClassicAddress(pubKey)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Wallet{
+		PublicKey:      pubKey,
+		PrivateKey:     fmt.Sprintf("00%s", privKey),
```
Comment (florent-uzio) [inline xrpl/wallet.go:117]: So this generation from a mnemonic only generates a secp256k1 key ? Is it possible to generate an ed25519 also ?

**Review by florent-uzio — APPROVED** [review]
Looks great to me! Nice job!

File: .github/workflows/xrpl-go-ci.yml (hunk 1 above)
Comment (GuillemGarciaDev) [reply .github/workflows/xrpl-go-ci.yml:6]: Its a pattern to reference all branch names

File: xrpl/wallet.go (hunk 2 above)
Comment (GuillemGarciaDev) [reply xrpl/wallet.go:117]: By default it should be secp256k1 to match the rippled `wallet_propose` default algorithm. If needed we can add it later

**Pull request merged**